packages/toolkit/src/query/core/buildThunks.ts

```diff
@@ -430,6 +433,8 @@ In the case of an unhandled error, no tags will be "provided" or "invalidated".`
   const hasMaxAge = (
     options: any
   ): options is { ifOlderThan: false | number } => 'ifOlderThan' in options
+  const handlePrefetchSubscription = (result: QueryActionCreatorResult<any>) =>
+    result.unwrap().finally(result.unsubscribe)
```

Comment:
I opted to `unwrap` because that seems closer aligned to the lifecycle the end-user is expecting. My hope is it also sidesteps worrying about `keepUnusedDataFor` being set to 1 second or other cache edge-cases I can't think of.